kitsune/search/v2/es7_utils.py

```diff
@@ -148,15 +150,24 @@ def index_objects_bulk(doc_type_name, obj_ids, timeout=settings.ES_BULK_DEFAULT_
     # if the request doesn't resolve within `timeout`,
     # sleep for `timeout` then try again up to `settings.ES_BULK_MAX_RETRIES` times,
     # before raising an exception:
-    es7_bulk(
-        es7_client(
-            timeout=timeout,
-            retry_on_timeout=True,
-            initial_backoff=timeout,
-            max_retries=settings.ES_BULK_MAX_RETRIES,
-        ),
-        (doc.to_action(action=action, is_bulk=True, **kwargs) for doc in docs),
-    )
+    try:
+        es7_bulk(
+            es7_client(
+                timeout=timeout,
+                retry_on_timeout=True,
+                initial_backoff=timeout,
+                max_retries=settings.ES_BULK_MAX_RETRIES,
+            ),
+            (doc.to_action(action=action, is_bulk=True, **kwargs) for doc in docs),
+        )
+    except BulkIndexError as exc:
+        exc.errors[:] = [
+            error
+            for error in exc.errors
+            if (error.get("delete") and error["delete"]["status"] not in [400, 404])
+        ]
+        if exc.errors:
+            raise exc
```

Comment:
This is :chefs-kiss-emoji:, though a comment explaining this is modifying in place might be nice (since it's pretty obscure syntax).

I'm not sure the if statement is quite right, though:
if `error = { "update": {} }`, then the statement returns `None` and that error would be excluded from the array - when we want it to be included and raised.
I think the if statement needs to be:
```
if not (error.get("delete") and error["delete"]["status"] in [400, 404])
```